### PR TITLE
Automatically deploy pkgdown page to branch gh-page

### PR DESCRIPTION
**Pull request by christianholland**
I added GitHub actions that automatically checks the package on every push or pull request. After success the pkgdown page will be automatically updated and deposited at a separate branch named gh-pages

Please note that currently the checks fail due to incompatible R versions, however after the release of R4.1 (May 20th) it will start working

### CODE REVIEW

**Comment by christianholland** [comment]
I forgot to revert the R version back to 4.1 that's why there is a conflict in the DESCRIPTION file

File: README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # cosmosR <img src="man/figures/logo.png" align="right" height="139">
 
 <!-- badges: start -->
+[![R-CMD-check](https://github.com/saezlab/dorothea/workflows/R-CMD-check-bioc/badge.svg)](https://github.com/saezlab/cosmosr/actions)
```
Comment (gabora) [inline README.md]: why do we need to use dorothea repo here?

Comment (gabora) [reply README.md]: I just saw your next commit :D

**Review by gabora — APPROVED** [review]
better later than never :)
thanks for the contribution, finally I got time to understand whats going on here:)

**Comment by gabora** [comment]
integrated your PR locally, thanks again!

**Pull request closed without merging**